https://github.com/FluidSynth/fluidsynth/pull/1244
From: Brahmajit Das <brahmajit.xyz@gmail.com>
Date: Thu, 11 May 2023 23:14:42 +0530
Subject: [PATCH] Fix implicit declaration of function strcasecmp with musl

This will also help in making the source code c99 compilant, thus
helping in packaing for newer compilers (for example clang 16 and gcc 14)

Signed-off-by: Brahmajit Das <brahmajit.xyz@gmail.com>
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -159,6 +159,7 @@ include ( CheckIncludeFile )
 include ( CheckSymbolExists )
 include ( CheckTypeSize )
 check_include_file ( string.h HAVE_STRING_H )
+check_include_file ( strings.h HAVE_STRINGS_H )
 check_include_file ( stdlib.h HAVE_STDLIB_H )
 check_include_file ( stdio.h HAVE_STDIO_H )
 check_include_file ( math.h HAVE_MATH_H )
--- a/src/utils/fluidsynth_priv.h
+++ b/src/utils/fluidsynth_priv.h
@@ -45,6 +45,9 @@
 #include <string.h>
 #endif

+#if HAVE_STRINGS_H
+#include <strings.h>
+#endif

 #include "fluidsynth.h"

--
2.40.1

